src/azure-cli/azure/cli/command_modules/appconfig/_utils.py

```diff
@@ -6,41 +6,51 @@
+from ._constants import HttpHeaders
+
+logger = get_logger(__name__)
 
 
 def construct_connection_string(cmd, config_store_name):
     connection_string_template = 'Endpoint={};Id={};Secret={}'
+    # If the logged in user/Service Principal does not have 'Reader' or 'Contributor' role
+    # assigned for the requested AppConfig, resolve_resource_group will raise CLI error
+    resource_group_name, endpoint = resolve_resource_group(cmd, config_store_name)
 
     try:
-        resource_group_name, endpoint = resolve_resource_group(
-            cmd, config_store_name)
         config_store_client = cf_configstore(cmd.cli_ctx)
         access_keys = config_store_client.list_keys(
             resource_group_name, config_store_name)
         for entry in access_keys:
             if not entry.read_only:
                 return connection_string_template.format(endpoint, entry.id, entry.value)
-    except Exception:
+    except Exception as ex:
         raise CLIError(
-            'Cannot find the App Configuration {}. Check if it exists in the subscription that logged in. '.format(config_store_name))
+            'Failed to get access keys for the App Configuration "{}". Make sure that the account that logged in has "Contributor" role assigned for this App Configuration store.\n{}'.format(config_store_name, str(ex)))
 
     raise CLIError('Cannot find a read write access key for the App Configuration {}'.format(
         config_store_name))
 
 
 def resolve_resource_group(cmd, config_store_name):
-    config_store_client = cf_configstore(cmd.cli_ctx)
-    all_stores = config_store_client.list()
-    for store in all_stores:
-        if store.name.lower() == config_store_name.lower():
-            # Id has a fixed structure /subscriptions/subscriptionName/resourceGroups/groupName/providers/providerName/configurationStores/storeName"
-            return store.id.split('/')[4], store.endpoint
-    raise CLIError(
-        "App Configuration store: {} does not exist".format(config_store_name))
+    try:
+        config_store_client = cf_configstore(cmd.cli_ctx)
+        all_stores = config_store_client.list()
+        for store in all_stores:
+            if store.name.lower() == config_store_name.lower():
+                # Id has a fixed structure /subscriptions/subscriptionName/resourceGroups/groupName/providers/providerName/configurationStores/storeName"
+                return store.id.split('/')[4], store.endpoint
+    except ErrorException as ex:
+        raise CLIError("Failed to get the list of App Configuration stores for the current user. Make sure that the account that logged in has 'Reader' or 'Contributor' role assigned for the required App Configuration store.\n{}".format(str(ex)))
+
+    raise CLIError("Failed to find the App Configuration store '{}'. Make sure that the account that logged in has 'Reader' or 'Contributor' role assigned for this App Configuration store.".format(config_store_name))
 
```

Comment:
I assume ErrorException indicates the permsision issue? If it throws other error, it has nothing to do with role.

Reply:
ErrorException is only thrown if the logged in account doesn't have access to any config store, i.e. config_store_client.list() returns no config stores.
It's possible that the logged in account has access to some stores, but not the one they are trying to access here. In that case, we need to raise CLI error if the return statement on line 49 was never hit.

---
In reply to: [494700304](https://github.com/Azure/azure-cli/pull/15160#discussion_r494700304) [](ancestors = 494700304)

Reply:
You are right. But it is hard to guess the reason, like it can be no access or the store does not exist. 

---
In reply to: [494716795](https://github.com/Azure/azure-cli/pull/15160#discussion_r494716795) [](ancestors = 494716795,494700304)